auth/auth.go

```diff
@@ -0,0 +1,61 @@
+package auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+)
+
+type AccessToken struct {
+	AccessToken string `json:"access_token"` // Valid for 6hrs
+	// TODO: Use the below value to determine whether or not to make the request
+	ExpiresIn   int32  `json:"expires_in"`
+}
+
+func GetToken(apiUrl, clientId, clientSecret string) (tkn *AccessToken, err error) {
+
+	if len(clientId) == 0 && len(clientSecret) == 0 {
+		log.Fatal(" Both CLIENT_ID and CLIENT_SECRET environment variables are required")
+	}
```

Comment:
See, if you blew up early on you could avoid this check (assuming you trust the main to), plus this `auth` package shouldn't really know about these two environment variables ☺️